### ISSUE TITLE
Close modal onConfirm and onCancel

### ISSUE THREAD

**Issue by chermoso** [henninghall/react-native-date-picker#609]
**Describe the bug**
When I select a date or cancel the picker, it closes not only de datepicker modal but also a modal that wraps it, opened before.

**Expected behavior**
It just have to close the picker modal.

**To Reproduce**
Here's an imcomplete example, just to show you whats the point.
```javascript
export default const App = () => {
  const [date, setDate] = useState()
  const [isOpened, setDate] = useState()
  return <>
      <Modal open={isOpened}>
        <DatePicker
          modal
          date={this.state.date}
          onConfirm={date => this.setState({ date })}
        />
      </Modal>
    </>

}
```

**Comment by chermoso** [comment]
I guess is related to this commit
https://github.com/henninghall/react-native-date-picker/commit/c0d6fe934555f7aedcde835d831590ae1ab52722

**Comment by sarahabuteen** [comment]
Same behaviour occurs when using react-navigation, if the screen presentation is set to "modal", it closes the whole screen.

**Comment by henninghall** [comment]
Thanks for reporting this! it is fixed and released in version 4.2.9
